packages/reporting-server/rest_server/__main__.py

```diff
@@ -1,18 +1,48 @@
-import uvicorn
+import asyncio
 
-from .app import app
-from .app_config import app_config
+from uvicorn import Config, Server
 
+from .app import get_app
+from .app_config import SystemMode, app_config
 
-def main():
-    uvicorn.run(
-        app,
+
+class MyServer(Server):
+    # pylint: disable=invalid-overridden-method
+    async def run(self, sockets=None):
+        self.config.setup_event_loop()
+        return await self.serve(sockets=sockets)
+
+
+async def run():
+    apps = []
+    config1 = Config(
+        get_app(SystemMode.FLUENTD),
+        host=app_config.host,
+        port=app_config.port_fluentd,
+        root_path=app_config.public_url.path,
+        log_level=app_config.log_level.lower(),
+    )
+    config2 = Config(
+        get_app(SystemMode.REPORT),
         host=app_config.host,
         port=app_config.port,
         root_path=app_config.public_url.path,
         log_level=app_config.log_level.lower(),
     )
 
+    server1 = MyServer(config=config1)
+    server2 = MyServer(config=config2)
+
+    apps.append(server1.run())
+    apps.append(server2.run())
+
+    return await asyncio.gather(*apps)
+
+
+def main():
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(run())
```

Comment:
Downside of this is that uvicorn auto reload no longer works? I'm not sure if there is any way to get the best of both worlds except totally separating the 2 components, but then you can't run both components at once like this.

Reply:
Yeah, if you run with `reporting_server` command it will not reload, but if we use this command `uvicorn --reload rest_server.app:get_app` for development, it will reload the app without any problems, because it'll run one instance of uvicorn (and you can configure what you want to expose in that one instance, all, report or fluentd).